app-codepipeline/vars.tf

```diff
@@ -8,6 +8,17 @@ variable "name_prefix" {
   }
 }
 
+variable "sdlc_stage" {
```

Comment:
Also fine for current PR, but remember that the pipelines are going to span multiple environments.  Rather than having a set SDLC stage/environment, the config will look something like this:

```
stages = {
  # build, test, and push images to dev ECR
  dev = {
    # environment, aka sdlc_stage
    # could be redundant if we enforce stage id = env name
    environment: dev

    # define the actions for this stage
    actions = {
      # Define an action named "build_each" that builds each service in parallel
      build_each = [
        {
          name: "Build Public Site"
          role: <role_arn>
          buildspec: "path/to/buildspec.yaml"
        },
        {
          name: "Build Partner Site"
          role: <role_arn>
          buildspec: "path/to/buildspec.yaml"
        },
        {
          name: "Build Backend"
          role: <role_arn>
          buildspec: "path/to/buildspec.yaml"
        }
      ]

      # Define an action named "deploy" that deploys changes if build_each is successful
      deploy = {
        name: "Deploy "
        role: <role_arn>
        buildspec: "path/to/buildspec.yaml"
      }
    }
  }

  staging = {
    environment: staging
    ...
  }

  prod = {
    environment: prod
    ...
  }
}
```

That makes it easier to add new actions (ie add a new container to build) and reduces the amount of TF code needed.  It also makes it easier for the devs that inherit this to better understand what's going on, even though they won't have as much TF knowledge to start, and for others who might want to use this (this is an open source project) to be able to modify the pipeline without having to change the actual code itself, potentially creating a hard fork.

Reply:
ok I was under the impression we would have a `dev.tfvars`, `staging.tfvars`, `prod.tfvars` in the config repo. IAM permissions are tied to each set of vars and terraform workspace via https://github.com/metrotranscom/doorway-infra/issues/99